### DISCUSSION TITLE
Tables with curves dont show up as tables

### DISCUSSION THREAD

**Discussion by joedimagio** [jsvine/pdfplumber#1052] [Ask for help with specific PDFs]
Fetching the tables shows me this

![img4](https://github.com/jsvine/pdfplumber/assets/5310515/1e7820e2-bf4b-4c67-916f-a734b52be182)

Table debug shows me this

![img5](https://github.com/jsvine/pdfplumber/assets/5310515/52dc811a-c73a-4dbd-9173-532d3b89ac05)

Fetching the curves shows me this

![img3](https://github.com/jsvine/pdfplumber/assets/5310515/f6647e3c-b4c4-4e22-a5fc-62d3e4e6e4d2)

I am using version 0.10.2 and assumed that curves are incorporated into table extraction and hence the table extraction should be good.  What am I missing?

**Comment by jsvine** [comment]
Curves are not included by default, but you can use them by passing them to "explicit_vertical_lines" and "explicit_horizontal_lines". E.g.:

```python
tables = page.extract_tables({
  "explicit_vertical_lines": page.edges + page.curves,
  "explicit_horizontal_lines": page.edges + page.curves,
})
```

For your PDF this may have some unintended consequences, due to the fact that some of the "text" appears to be defined as curves (rather than proper text). You might be able to handle this via `page.filter(...)`, if you can identify some characteristics of the `.curve` objects that are different for the text-drawing curves vs. the table-drawing curves.

**Reply by joedimagio** [reply]
Thanks for the suggestion. With explicit lines as suggested, this is what I get

![table-edges](https://github.com/jsvine/pdfplumber/assets/5310515/fa4fe2ec-0680-44bc-8f7a-01c18e614a14)

However just drawing the edges (im.draw_lines(page.edges, stroke="red", stroke_width=2) gives me the following

![edges](https://github.com/jsvine/pdfplumber/assets/5310515/68e724e7-026d-4f4d-93f6-bb554bb6c19f)

The table extraction has a couple of vertical lines which are off. Any clues on why this happens or how to fix it?

[sampleR.pdf](https://github.com/jsvine/pdfplumber/files/13563189/sampleR.pdf)

**Reply by jsvine** [reply]
There seems to be some strange interaction between various line-line elements on the page, but this seems to work for me:

```python
im.reset().debug_tablefinder({
    "horizontal_strategy": "explicit",
    "vertical_strategy": "explicit",
    "explicit_horizontal_lines": page.curves,
    "explicit_vertical_lines": page.curves + [ x for x in page.lines if 5 < x["height"] < 100],
})
```

![image](https://github.com/jsvine/pdfplumber/assets/534702/d5e0a85d-df38-405e-bc96-3ea1eacbcb06)

**Reply by joedimagio** [reply]
Thank you, that works.
